src/main/java/org/broadinstitute/hellbender/tools/exome/CnvCaller.java

```diff
@@ -27,6 +28,8 @@
  *
  * Note:  Assumes that the incoming target file is in log_2(CR)
  * Note:  Assumes that the incoming segment means are CR.
+ * Note: Assumes that the input @{link ReadCountCollection} contains only one sample -- calls will be made for only one
```

Comment:
I agree it's a code smell but I don't think the smell is that the underlying value is a vector but rather that the code takes in this inherently matrix quality (the `ReadCountCollection` of multiple samples) and refuses to work properly if there is more than one sample.  One overall smell in our codebase involving the redundancy between `TargetCoverage` and `ReadCountCollection` was that some tools *in the same pipeline* worked fine with multiple samples and some didn't.

In my opinion the solution would be to rewrite this tool and others to call multiple samples at once, just as the switch to `ReadCountCollection` now trivially makes `NormalizeSomaticReadCounts` work for multiple samples.

As for the name, I totally agree.  What do you think about renaming it the integer/double-agnostic `CoverageCollection`?